lib/launchers/ocm.rb

```diff
@@ -0,0 +1,197 @@
+      if @cloud
+        @cloud_opts = default_opts(@cloud)
+        unless @cloud_opts
+          raise "Cannot find cloud '#{cloud}' defined in '#{service_name}'"
+        end
+      end
+    end
+
+    # @param service_name [String] the service name of this openstack instance
+    #   to lookup in configuration
+    def default_opts(service_name)
+      return  conf[:services, service_name.to_sym]
+    end
+
+    def to_seconds(string)
+      regex_m = /^(\d+)\s*(m|min|minutes|mins)+$/
+      regex_h = /^(\d+)\s*(h|hour|hours|hrs)+$/
+      regex_d = /^(\d+)\s*(d|day|days)+$/
+      regex_w = /^(\d+)\s*(w|week|weeks|wks)+$/
+      if string.match(regex_m)
+        return string.match(regex_m)[1].to_i * 60
+      elsif string.match(regex_h)
+        return string.match(regex_h)[1].to_i * 60 * 60
+      elsif string.match(regex_d)
+        return string.match(regex_d)[1].to_i * 24 * 60 * 60
+      elsif string.match(regex_w)
+        return string.match(regex_w)[1].to_i * 7 * 24 * 60 * 60
+      else
+        raise "Cannot convert '#{string}' to seconds!"
+      end
+    end
+
+    # create a json which specifies OSD cluster
+    # in the future we plan to move the logic into the script 'osd-provision.sh'
+    def generate_json(name)
+      json_data = {
+        "name" => name,
+        "managed" => true,
+        "multi_az" => false,
+        "byoc" => false
+      }
+
+      if @multi_az
+        json_data.merge!({"multi_az" => @multi_az})
+      end
+
+      if @region
+        json_data.merge!({"region" => {"id" => @region}})
+      end
+
+      if @version
+        json_data.merge!({"version" => {"id" => "openshift-v#{@version}"}})
+      end
+
+      if @nodes
+        json_data.merge!({"nodes" => {"compute" => @nodes.to_i}})
+      end
+
+      if @lifespan
+        expiration = Time.now + to_seconds(@lifespan)
+        json_data.merge!({"expiration_timestamp" => expiration.strftime("%Y-%m-%dT%H:%M:%SZ")})
+      end
+
+      if @cloud_opts
+        case @cloud_opts[:cloud_type]
+        when "aws"
+          aws = Amz_EC2.new(service_name: @cloud)
+          json_data.merge!({"aws" => {"access_key_id":aws.access_key, "secret_access_key":aws.secret_key, "account_id":aws.account_id}})
+          json_data.merge!({"byoc" => true})
+        end
+      end
+
+      return json_data.to_json
+    end
+
+    # download the script 'osd-provision.sh' which takes care of the OSD installation/uninstallation
+    def download_osd_script
+      default_osd_script_url = "https://gitlab.cee.redhat.com/mk-bin-packing/mk-performance-tests/-/raw/master/scripts/osd-provision.sh?inline=false"
+      osd_script_url = ENV['OSD_SCRIPT_URL'] || default_osd_script_url
+      # we need another parent dir as the script downloads ocm tool to its parent
+      osd_scripts = "#{Dir.tmpdir}/osd-scripts"
+      %x(
+        rm -rf #{osd_scripts} && mkdir #{osd_scripts} && \
+        curl -s #{osd_script_url} --output #{osd_scripts}/osd-provision.sh && \
```

Comment:
Please never use `curl` and other native commands unless absolutely necessary and after discussion. Please see how to download file example in step `I download a file from` inside `features/step_definitions/net.rb` using method `BushSlicer::Http.get`. You can chmod file like `File.chmod(0600, ssh_key)`.

Reply:
I wasn't aware of the `BushSlicer::Http.get` so I will use it.

Reply:
I have changed it to cloning full git repo.